ext/ManoptLRUCacheExt.jl

```diff
@@ -0,0 +1,93 @@
+module ManoptLRUCacheExt
+
+using Manopt
+import Manopt: init_caches
+using ManifoldsBase
+
+if isdefined(Base, :get_extension)
+    using LRUCache
+else
+    # imports need to be relative for Requires.jl-based workflows:
+    # https://github.com/JuliaArrays/ArrayInterface.jl/pull/387
+    using ..LRUCache
+end
+
+# introduce LRU even as default.
+function Manopt.init_caches(
+    M::AbstractManifold, caches::AbstractVector{<:Symbol}; kwargs...
+)
+    return Manopt.init_caches(M, caches, LRU; kwargs...)
+end
+
+"""
+    init_caches(caches, T::Type{LRU}; kwargs...)
+
+Given a vector of symbols `caches`, this function sets up the
+`NamedTuple` of caches, where `T` is the type of cache to use.
+
+# Keyword arguments
+
+* `p`   - (`rand(M)`) a point on a manifold, to both infere its type for keys and initialize caches
+* `v`   - (`0.0`) a value both typing and initialising number-caches, eg. for caching a cost.
```

Comment:
I think I'd prefer a bit more verbose name, maybe `value`?

Reply:
do you have a preference? It is the value and type o initialise cost or constraint values. So besides points and tangents, we cache arbitrary values – hence the name

Reply:
I don't really have much preference other than that it should be something longer.

Reply:
Hm, and I am not creative enough to provide something besides `value=` – I thought `v` would be fine besides `p` and `X` though.

Reply:
`p` and `X` are used commonly in the ecosystem so it's an established convention, `v` not so much. Anyway, if you prefer `v` over `value`, then let's keep `v`.

Reply:
We can also switch to value, since you are right, `v` is not so usual.